.travis.yml

```diff
@@ -1,16 +1,53 @@
 language: rust
-rust:
-  - nightly
-  - beta
-  - stable
-env:
-  - GTK=3.4
-  - GTK=3.18
+matrix:
+  include:
+    - os: linux
+      rust: nightly
+      env: GTK=3.4
+    - os: linux
+      rust: nightly
+      env: GTK=3.18
+    - os: linux
+      rust: beta
+      env: GTK=3.4
+    - os: linux
+      rust: beta
+      env: GTK=3.18
+    - os: linux
+      rust: stable
+      env: GTK=3.4
+    - os: linux
+      rust: stable
+      env: GTK=3.18
+    - os: osx
+      rust: nightly
+      env: GTK=3.4
+    - os: osx
+      rust: nightly
+      env: GTK=3.18
+    - os: osx
+      rust: beta
+      env: GTK=3.4
+    - os: osx
+      rust: beta
+      env: GTK=3.18
+    - os: osx
+      rust: stable
+      env: GTK=3.4
+    - os: osx
+      rust: stable
+      env: GTK=3.18
+sudo: true
 addons:
   apt:
     packages:
     - libgtk-3-dev
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew unlink python; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install gtk+3 cairo atk; fi
 script:
+  - cargo doc --features "dox"
   - cargo test
   - rustc --version
```

Comment:
Maybe this line better removed as it already printed after rust installation
or moved as first line